tests/http/federation/test_matrix_federation_agent.py

```diff
@@ -75,8 +92,12 @@ def setUp(self):
 
         self.tls_factory = FederationPolicyForHTTPS(config)
 
-        self.well_known_cache = TTLCache("test_cache", timer=self.reactor.seconds)
-        self.had_well_known_cache = TTLCache("test_cache", timer=self.reactor.seconds)
+        self.well_known_cache: TTLCache[bytes, Optional[bytes]] = TTLCache(
+            "test_cache", timer=self.reactor.seconds
+        )
+        self.had_well_known_cache: Optional[TTLCache[bytes, bool]] = TTLCache(
+            "test_cache", timer=self.reactor.seconds
+        )
```

Comment:
Is this really optional?

Reply:
Err, I don't think so. I probably just copied this https://github.com/matrix-org/synapse/blob/17e1eb7749adf12d43f534c50115bbe19c809ea6/synapse/http/federation/well_known_resolver.py#L91

without thinking.

Reply:
85afaef92c fixes this.